x/stake/handler.go

```diff
@@ -34,8 +34,16 @@ func NewHandler(k keeper.Keeper) sdk.Handler {
 func EndBlocker(ctx sdk.Context, k keeper.Keeper) (ValidatorUpdates []abci.ValidatorUpdate) {
 	endBlockerTags := sdk.EmptyTags()
 
+	// reset the intra-transaction counter
+	k.SetIntraTxCounter(ctx, 0)
+
+	// Calculate validator set changes
+	ValidatorUpdates = k.ApplyAndReturnValidatorSetUpdates(ctx)
```

Comment:
Does this ordering change fix some bug or is it just cleaner? If it does fix a bug, can we add a comment as to why this order must be preserved? As far as I know validator update calculation and unbonding the queues should be non-interfering (but maybe I'm wrong).

Reply:
It fixes a bug when the unbonding period is instant (is the case in some of the tests).  The test expected the validator to be completely unbonded after the Endblocker (go from Bonded -> Unbonding during `ApplyAndReturnValidatorSetUpdates` and then Unbonding -> Unbonded during `UnbondAllMatureValidatorQueue`).  For this reason, `ApplyAndReturnValidatorSetUpdates` has to come before `UnbondAllMatureValidatorQueue`.